redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/YogaUIView.kt

```diff
@@ -7,42 +7,79 @@
 package app.cash.redwood.layout.uiview
 
 import app.cash.redwood.layout.api.Constraint
+import app.cash.redwood.yoga.FlexDirection
 import app.cash.redwood.yoga.MeasureCallback
 import app.cash.redwood.yoga.MeasureMode
 import app.cash.redwood.yoga.Node
 import app.cash.redwood.yoga.Size
 import kotlinx.cinterop.CValue
+import kotlinx.cinterop.ObjCClass
 import kotlinx.cinterop.cValue
 import kotlinx.cinterop.useContents
 import platform.CoreGraphics.CGRectMake
 import platform.CoreGraphics.CGRectZero
 import platform.CoreGraphics.CGSize
 import platform.CoreGraphics.CGSizeMake
+import platform.UIKit.UILabel
+import platform.UIKit.UIScrollView
+import platform.UIKit.UIScrollViewContentInsetAdjustmentBehavior
 import platform.UIKit.UIView
 import platform.UIKit.UIViewNoIntrinsicMetric
 
-internal class YogaUIView : UIView(cValue { CGRectZero }) {
+internal class YogaUIView : UIScrollView(cValue { CGRectZero }) {
   val rootNode = Node()
 
   var width = Constraint.Wrap
   var height = Constraint.Wrap
 
-  var applyModifier: (Node, Int) -> Unit = { _, _ -> }
+  var applyModifier: (Node, Int) -> Unit = { _, _ -> println("applyModifier")}
+
+  init {
+    // TODO: Support Scroll Indicators
+    showsVerticalScrollIndicator = false
+    showsHorizontalScrollIndicator = false
+    contentInsetAdjustmentBehavior = UIScrollViewContentInsetAdjustmentBehavior.UIScrollViewContentInsetAdjustmentNever
+  }
 
   override fun intrinsicContentSize(): CValue<CGSize> {
     return calculateLayoutWithSize(Size(Size.Undefined, Size.Undefined)).toCGSize()
   }
 
   override fun sizeThatFits(size: CValue<CGSize>): CValue<CGSize> {
-    val bounds = size.useContents { sizeToBounds(this) }
-    val output = calculateLayoutWithSize(bounds)
-    return output.toCGSize()
+    val constrainedSize = size.useContents { sizeForConstraints(this) }
+    val calculatedSize = calculateLayoutWithSize(constrainedSize)
+    return calculatedSize.toCGSize()
   }
 
   override fun layoutSubviews() {
-    val bounds = bounds.useContents { sizeToBounds(size) }
-    calculateLayoutWithSize(bounds)
+    super.layoutSubviews()
+
+    // Based on the constraints of Fill or Wrap, we
+    // calculate a size that the container should fit in.
+    val bounds = bounds.useContents {
+      sizeForConstraints(size)
+    }
 
+    if (scrollEnabled) {
+      // When scrolling is enabled, we want to calculate and apply the contentSize
+      // separately and have it grow a much as needed in the flexDirection.
+      // This duplicates the calculation we're doing above, and should be
+      // combined into one call
```

Comment:
Going to make a few of these suggestions, as [comments should be sentences](https://nedbatchelder.com/blog/201401/comments_should_be_sentences.html).
```suggestion
      // When scrolling is enabled, we want to calculate and apply the contentSize
      // separately and have it grow a much as needed in the flexDirection.
      // This duplicates the calculation we're doing above, and should be
      // combined into one call.
```
I think this is pretty uncontroversial, but it's something that people may have never thought about so it can sometimes be weird to get started thinking about it.